plugins/inputs/webhooks/common/basic_auth.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"crypto/subtle"
+	"net/http"
+)
+
+type BasicAuth struct {
+	Username string
+	Password string
+}
+
+func (b *BasicAuth) Verify(r *http.Request) bool {
+	if b.Username == "" && b.Password == "" {
+		return true
+	}
+
+	username, password, ok := r.BasicAuth()
+	return ok &&
+		subtle.ConstantTimeCompare([]byte(username), []byte(b.Username)) == 1 &&
+		subtle.ConstantTimeCompare([]byte(password), []byte(b.Password)) == 1
```

Comment:
you will want to make sure both of these execute, not just one, otherwise a timing attack can easily figure out which usernames are valid